app/jobs/smtp_job.rb

```diff
@@ -0,0 +1,33 @@
+class SMTPJob < ApplicationJob
+  queue_as :default
+
+  def perform(mail_alias, enable)
+    if enable
+      enable_smtp(mail_alias)
+    else
+      disable_smtp(mail_alias)
+    end
+  end
+
+  private
+
+  def enable_smtp(mail_alias)
+    password = SecureRandom.hex(16)
+    mailgun_client.post("/domains/#{mail_alias.domain}/credentials",
+                        'login': mail_alias.email, 'password': password)
+    MailSMTPMailer.enabled_email(mail_alias, password).deliver_later
+  end
+
+  def disable_smtp(mail_alias)
+    mailgun_client.delete("/domains/#{mail_alias.domain}/credentials/#{mail_alias.email}")
+    MailSMTPMailer.disabled_email(mail_alias).deliver_later
+  end
+
+  # :nocov:
+  def mailgun_client
+    api_key = Rails.application.config.x.mailgun_api_key
+    api_host = Rails.application.config.x.mailgun_host
+    @mailgun_client = Mailgun::Client.new api_key, api_host
```

Comment:
You don't need to store this in an instance variable when doing it this way.

If you want memoization, you should do this between line 27 and 28

Reply:
Why don't we need instance variables? If we don't do that it will create a new instance every call right?

Reply:
With the current code you are creating a new instance every call anyway. You assign it to `@mailgun_client` but it won't be skipped when it already exists. This is how you would do proper memoization:

```suggestion
    @mailgun_client ||= Mailgun::Client.new api_key, api_host
```

However, you can also skip the previous two lines when it exists, so maybe this:

```ruby
def mailgun_client
    return @mailgun_client unless @mailgun_client.nil?
    ...
    @mailgun_client = Mailgun::Client.new...
end
```